src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -42,49 +48,63 @@ export const selectStakingDataByAccountSpecifier = createSelector(
   }
 )
 
-export const selectDelegationCryptoAmountByDenom = createSelector(
+export const selectDelegationCryptoAmountByAssetId = createSelector(
   selectStakingDataByAccountSpecifier,
   selectValidatorAddress,
-  selectDenom,
-  (stakingData, validatorAddress, denom): string | undefined => {
+  selectAssetId,
+  (stakingData, validatorAddress, selectedAssetId): string | undefined => {
     if (!stakingData || !stakingData.delegations?.length) return
 
     const delegation = stakingData.delegations.find(
       ({ assetId, validator }) =>
-        ASSET_ID_TO_DENOM[assetId] === denom && validator.address === validatorAddress
+        assetId === selectedAssetId && validator.address === validatorAddress
     )
     return delegation?.amount
   }
 )
 
+export const selectAllDelegationsCryptoAmountByAssetId = createSelector(
+  selectStakingDataByAccountSpecifier,
+  selectAssetId,
+  (stakingData, selectedAssetId): amountByValidatorAddressType => {
+    if (!stakingData || !stakingData.delegations?.length) return {}
+
+    const delegations = stakingData.delegations.reduce(
+      (acc: amountByValidatorAddressType, { assetId, amount, validator: { address } }) => {
+        if (assetId !== selectedAssetId) return acc
+
+        acc[address] = amount
+        return acc
+      },
+      {}
+    )
+    return delegations
+  }
+)
+
 export const selectRedelegationEntriesByAccountSpecifier = createDeepEqualOutputSelector(
   selectStakingDataByAccountSpecifier,
   selectValidatorAddress,
-  (stakingData, validatorAddress): Array<{ denom: string; amount: string }> => {
+  (stakingData, validatorAddress): chainAdapters.cosmos.RedelegationEntry[] => {
     if (!stakingData || !stakingData.redelegations?.length) return []
 
     const redelegation = stakingData.redelegations.find(
       ({ destinationValidator }) => destinationValidator.address === validatorAddress
     )
 
-    return (
-      redelegation?.entries.map(redelegationEntry => ({
-        denom: ASSET_ID_TO_DENOM[redelegationEntry.assetId],
-        amount: redelegationEntry.amount
-      })) || []
-    )
+    return redelegation?.entries || []
   }
 )
 
-export const selectRedelegationCryptoAmountByDenom = createSelector(
+export const selectRedelegationCryptoAmountByAssetId = createSelector(
   selectRedelegationEntriesByAccountSpecifier,
-  selectDenom,
-  (redelegationEntries, denom): string | undefined => {
+  selectAssetId,
+  (redelegationEntries, selectedAssetId): string | undefined => {
```

Comment:
```suggestion
  (redelegationEntries, selectedAssetId): string => {
```

return `'0'` if zero

Reply:
done